### ISSUE TITLE
http directives

### ISSUE THREAD

**Issue by michaelstaib** [ChilliCream/hotchocolate#288]
In order to make it simple to integrate rest-apis into a GraphQL schema we want to create a set of directives that do that without any code needed. 

```GraphQL
type Query {
  hello: String @http(method:GET url:"http://foo/hello") @json(path:"$.hello")
}
```

**Comment by michaelstaib** [comment]
```graphql
type Query {
  hello(helloId: String): String @http(method:GET url:"http://foo/hello/{helloId}") @json(path:"$.hello")
}
```

**Comment by rstaib** [comment]
```graphql
type Query {
  person(id: ID): Person @http(method:GET url:"http://foo/person/{id}")
}

type Person {
  name: String @json('$.name')
}
```

**Comment by rstaib** [comment]
**Specific mapping**

```graphql
type Query {
  person(id: ID): Person @http(method:GET url:"http://foo/person/{args.id}")
}

type Person {
  name: String @json(path:"$.name")
  email: String @json(path:"$.emailAddress")
}
```

**Auto mapping**

```graphql
type Query {
  person(id: ID): Person @http(method:GET url:"http://foo/person/{args.id}")
}

type Person @json {
  name: String
  email: String
}
```

**Mixed mapping**

```graphql
type Query {
  person(id: ID): Person @http(method:GET url:"http://foo/person/{args.id}")
}

type Person @json(path:"$.first") {
  name: String
  email: String @json(path:"$.emailAddress")
}
```

**Comment by maryalfred25** [comment]
Hello @michaelstaib ,

Correct me if i'm wrong is this thee way to stitch rest api in GraphQL gateway?

**Comment by michaelstaib** [comment]
This is now implemented with 12.7
